lib/compress/zstd_compress.c

```diff
@@ -1199,42 +1200,61 @@ static size_t ZSTD_resetCCtx_usingCDict(ZSTD_CCtx* cctx,
                             U64 pledgedSrcSize,
                             ZSTD_buffered_policy_e zbuff)
 {
+    /* We have a choice between copying the dictionary context into the working
+     * context, or referencing the dictionary context from the working context
+     * in-place. We decide here which strategy to use. */
+    /* TODO: pick reasonable cut-off size, handle ZSTD_CONTENTSIZE_UNKNOWN */
+    int attachDict = pledgedSrcSize <= 8 KB
+                  && cdict->cParams.strategy == ZSTD_fast
+                  && ZSTD_equivalentCParams(cctx->appliedParams.cParams,
```

Comment:
What if `ZSTD_equivalentCParams()==0` ?
Does that mean the first time the dictionary is used, the strategy is to copy cdict tables ?

Reply:
Correct. If we didn't, we would have to do a full reset of the working context anyways, which would be just as expensive. And accessing the dict from the working context is faster.